src/client.rs

```diff
@@ -593,12 +594,12 @@ impl<'a> Client<'a> {
     /// Returns the current client status
     pub fn get_cur_status(&mut self) -> &ClientState {
         // Check to see if the status changed
-        let new_status = self.core_res.try_recv();
+        let new_status = self.core_res.recv().now_or_never();
```

Comment:
Hmm, there is something tricky going on with this `try_recv` story: https://github.com/tokio-rs/tokio/issues/3350. It seems to be fine to use `recv().now_or_never()`, but we should keep an eye on it.